### PR TITLE
Add hyperparameter search for LLM fine-tuning jobs

### PR DESCRIPTION
**Pull request by OlegSudakov**
The fine-tuning autoconfigurator allows to conduct hyperparameter search for fine-tuning jobs on Slurm cluster. Similarly to usual grid search, it starts a sequence of jobs for different hyperparameter configurations, and analyses their results to find the best-performing one in terms of validation loss.

### CODE REVIEW

**Review by dimapihtar — APPROVED** [review]
LGTM. Thank you!
